File: Baekjoon/Class4/01.15650 N과 M (2).js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim();

const [n, m] = input.split(" ").map(Number);
const visited = new Array(n + 1).fill(false);
const track = [];
const out = [];

const backTrack = (start) => {
    if(track.length === m) { // 출력
        out.push(track.join(" "));
        return;
    }
    for(let i = start; i <= n; i++) {
        if(!visited[i]) {
            track.push(i); // 추가
            visited[i] = true;
            backTrack(i + 1);
            track.pop(); // 삭제
            visited[i] = false;
        }
    }
}

backTrack(1);
console.log(out.join("\n"));

File: Baekjoon/Class4/01.15650 N과 M (2).py
import sys
input = sys.stdin.readline
write = sys.stdout.write

n, m = map(int, input().split())
visited = [False] * (n + 1)
track = []

def back_track(start):
    if len(track) == m: #출력
        write(f"{" ".join(track)}\n")
        return
    for i in range(start, n + 1):
        if not visited[i]:
            track.append(str(i)) #추가
            visited[i] = True
            back_track(i + 1)
            track.pop() #삭제
            visited[i] = False

back_track(1)